runtime/lib/ttnn/operations/data_movement/permute.cpp

```diff
@@ -0,0 +1,31 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "permute.h"
+
+#include "tt/runtime/detail/logger.h"
+#include "ttnn/common/constants.hpp"
+
+#include <vector>
+
+namespace tt::runtime::ttnn::operations::data_movement {
+void run(const ::tt::target::ttnn::PermuteOp *op, ProgramContext &context) {
+  ProgramTensorPool &tensorPool = context.getTensorPool();
+
+  const ::ttnn::Tensor &in = tensorPool.at(op->in()->global_id());
+  DEBUG_ASSERT(in.is_allocated());
+
+  std::vector<int64_t> permutation(op->permutation()->begin(),
```

Comment:
Consider using smallvector

Reply:
I deliberately went with `std::vector` in the runtime, as the runtime is totally independent of the LLVM. In my opinion, runtime should be the only exception from "prefer the LLVM ADT to std structures" in our software stack.